symengine/derivative.cpp

```diff
@@ -438,7 +438,27 @@ static RCP<const Basic> diff(const CLASS &self, \
             }
             return UnivariateIntPolynomial::from_dict(self.get_var(), std::move(d));
         } else {
-            return zero;
+            map_uint_mpz d;
+            d[0] = integer_class(0);
+            return UnivariateIntPolynomial::from_dict(self.get_var(), std::move(d));
+        }
+    }
+
+    static RCP<const Basic> diff(const UnivariatePolynomial &self,
+            const RCP<const Symbol> &x) {
+        if (self.get_var()->__eq__(*x)) {
+            map_int_Expr d;
+            for (const auto &p : self.get_dict()) {
+                if (p.first - 1 >= 0) {
+                    d[p.first - 1] = p.second * p.first;
+                }
+            }
+            return make_rcp<const UnivariatePolynomial>(self.get_var(),
+                    (--(d.end()))->first, std::move(d));
+        } else {
+            map_int_Expr d;
+            d[0] = Expression(0);
```

Comment:
That's because the derivative of a constant (since we are differentiating by y, x is a constant w.r.t to y_ is zero.